Sources/Datadog/Core/System/CarrierInfoProvider.swift

```diff
@@ -48,7 +50,7 @@ internal protocol CarrierInfoProviderType {
 extension CarrierInfo.RadioAccessTechnology {
     init(ctRadioAccessTechnologyConstant: String) {
         switch ctRadioAccessTechnologyConstant {
-        #if !targetEnvironment(macCatalyst)
+        #if os(iOS) && !targetEnvironment(macCatalyst)
```

Comment:
Do we need that `!targetEnvironment(macCatalyst)` if we're checking specifically for iOS?

Reply:
Indeed, good catch 👍